File: app/src/main/java/work/petrichor/petrichor/modules/FaceDetectionTask.java
package work.petrichor.petrichor.modules;

import android.os.AsyncTask;

import java.util.ArrayList;
import java.util.List;

import work.petrichor.seeta.Detector;
import work.petrichor.seeta.FaceInfo;
import work.petrichor.seeta.ImageData;

/**
 * @author Mitscherlich.
 * @date 2017/7/26.
 **/

public class FaceDetectionTask extends AsyncTask<ImageData, Void, List<FaceInfo>> {

    private String modelPath;

    public FaceDetectionTask(String path) {
        modelPath = path;
    }

    @Override
    protected List<FaceInfo> doInBackground(ImageData[] frames) {
        long now = System.currentTimeMillis();

        Detector detector = new Detector(modelPath);

        List<FaceInfo> faces = detector.detect(frames[0]);

        long ms = System.currentTimeMillis() - now;
        if (faces == null || faces.isEmpty()) {
            android.util.Log.i("FaceDetectionTask", "None face detected in " + ms + "ms!");
            return new ArrayList<>();
        } else {
            android.util.Log.i("FaceDetectionTask", "Detected " + faces.size() +  " face(s) in " + ms + "ms!");
            return faces;
        }
    }
}

File: app/src/main/java/work/petrichor/petrichor/modules/FaceDetectionTask.kt
package work.petrichor.petrichor.modules

import android.os.AsyncTask
import work.petrichor.facedetection.detection.Detector
import work.petrichor.facedetection.format.FaceInfo
import work.petrichor.facedetection.format.ImageData

/**
 * @author  Mitscherlich.
 * @date    2017/7/25.
 * **/
class FaceDetectionTask(private val modelPath: String) : AsyncTask<ImageData, Void, Array<FaceInfo>>() {

    override fun doInBackground(vararg frame: ImageData?): Array<FaceInfo> {
        return processFrame(frame[0])
    }

    private fun  processFrame(frame: ImageData?): Array<FaceInfo> {
        if (frame == null || frame.frameData.isEmpty()) return emptyArray()
        val detector = Detector(modelPath)
        val faces = detector.detect(frame) ?: emptyArray()
        return if (faces.isNotEmpty()) faces else emptyArray()
    }
}